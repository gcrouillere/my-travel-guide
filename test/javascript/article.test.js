import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Article from '../../app/javascript/packs/article'
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

jest.mock('../../app/javascript/utils/ajaxHelpers')

describe('Article test Suite', () => {
  it('renders component', async () => {
    const ArticleTree = mount(<Article match={{ params: { id: 1}}}/>)
    ArticleTree.instance().createTextContentsHTML = jest.fn((textContents) => {
      return textContents.map(x => x.text).reduce((x, acc) => x + acc)
    })
    ArticleTree.update()
    await ArticleTree.instance().componentDidMount()
    ArticleTree.update()

    expect(toJson(ArticleTree)).toMatchSnapshot()
  })

})
