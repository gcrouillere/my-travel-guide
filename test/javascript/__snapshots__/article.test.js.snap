// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article test Suite renders component 1`] = `
<Article
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
>
  <div
    className="container article-container"
  >
    <header
      className="text-center"
    >
      <h1>
        My 10 characters long article title
      </h1>
    </header>
    <div
      id="articleTextContents"
    >
      Text content is not empty
    </div>
  </div>
</Article>
`;
